dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -425,21 +420,18 @@ public CloseableClient build() {
 
             setupConnectionPoolMetrics(conf.taggedMetricRegistry(), name, connectionManager);
 
+            boolean proxyConfigured = !ProxySelector.getDefault().equals(conf.proxy());
             HttpClientBuilder builder = HttpClients.custom()
                     .setDefaultRequestConfig(RequestConfig.custom()
-                            // HTTPCLIENT-1478: Work around the connection timeout being used to negotiate proxy
-                            // traffic.
-                            .setConnectTimeout(
-                                    ProxySelector.getDefault().equals(conf.proxy())
-                                            ? connectTimeout
-                                            : Timeout.ofMilliseconds(socketTimeoutMillis))
+                            // HTTPCLIENT-2091: Work around the connection timeout being applied to proxy traffic.
```

Comment:
Fixed the ticket number here, I pasted the wrong link originally...

Reply:
and removed entirely, no need to work around this at all when we set a RequestConfig.responseTimeout.